assets/js/base/components/quantity-selector/index.js

```diff
@@ -13,30 +19,121 @@ const QuantitySelector = ( {
 	className,
 	quantity = 1,
 	onChange = () => null,
+	itemName = '',
 } ) => {
-	const classes = classNames(
-		'wc-block-quantity-selector__input',
-		className
+	const classes = classNames( 'wc-block-quantity-selector', className );
+	const inputRef = useRef( null );
+	const [ currentValue, setCurrentValue ] = useState( quantity );
+
+	useEffect( () => {
+		if ( currentValue !== inputRef.current.value ) {
+			inputRef.current.value = currentValue;
+		}
+		onChange( currentValue );
+	}, [ currentValue, onChange ] );
+
+	/**
+	 * Handles keyboard up and down keys to change quantity value.
+	 *
+	 * @param {Object} event event data.
+	 */
+	const quantityInputOnKeyDown = useCallback(
+		( event ) => {
+			if ( event.keyCode === KEYCODE_UP ) {
+				event.preventDefault();
```

Comment:
Nitpicking, but if `event.preventDefault();` is called in both `if` clauses, maybe we could move it outside?

Reply:
> Nitpicking, but if `event.preventDefault();` is called in both `if` clauses, maybe we could move it outside?

the event could be anything, not just up and down events, by moving it out of the `if` clause, we will prevent all keys events.

Reply:
Oh, right! :man_facepalming: